collection_requirements.rst

```diff
@@ -0,0 +1,114 @@
+*****************************
+Ansible Collections Checklist
+*****************************
+
+.. contents:: Topics
+
+Overview
+========
+
+**Warning:** Subject to frequent updates
+       This is a "living document", expect it to change as we progress with the Collections work over the next few months.
+
+Feedback
+========
+
+As with any project it's very important that we get feedback from users, contributors and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.
+
+Please raise feedback by
+
+* Discussing in ``#ansible-community`` in Freenode IRC
+* Adding to the `Community Working Group IRC meeting <https://github.com/ansible/community/issues/539>`_
+* Creating `GitHub Issues <https://github.com/ansible-collections/overview/issues>`_ against this repo
+
+
+Why is this needed
+===================
+
+
+Repo structure
+===============
+
+galaxy.yml
+----------
+
+* tags MUST set
+* dependencies MUST be set to ``'>=1.0.0'`` (Might have to think about dependency order when publishing)
+
+meta/runtime.yml
+----------------
+Example: `meta/runtime.yml <https://github.com/ansible-collections/collection_template/blob/main/meta/runtime.yml>`_
+* MUST define the minimim version of Ansible which this collection works with
+  * If the Collection works with Ansible 2.9, then this shuld be set to
+
+Modules & Plugins
+------------------
+
+Documentation
+~~~~~~~~~~~~~~
+All module and plugin ``DOCUMENTATION`` MUST
+
+* Use the FQCN for ``M(...)`` FIXME Docs link
+* Use the FQCN for ``SEEALSO`` FIXME Docs link
+* Use the FQCN for ``extends_documentation_fragment:``, unles you are referring to doc_fragments from ansible-base
+
+Contributor Workflow
+====================
+
+Changelogs
+~~~~~~~~~~
+
+FIXME: What's the actual minimum requirement here, is it a file called X?
+
+FIXME: Add details and examples of automated system
+
+Versioning and deprecation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FIXME Details here
+* Deprecation vs Collection version
+* How many Collection major versions to give
```

Comment:
I'm not sure what you mean with the two points. Do we need anything else, except that they stick to semver?

Reply:
How many major versions do things need to stick around for before they can be deleted?

Reply:
Hmm, good question. I think we should expect collections to give at least a ~6 months deprecation warning (i.e. one Ansible release cycle). So if they release major versions once every month, they should have a deprecation period of six major versions, but if they release only once or twice per year, a deprecation period of one major verison is fine.

(Deprecation period of one major version for me means that that there is at least one x.0.0 release where the thing is deprecated, and it is removed in (x+1).0.0. So if it is deprecated in x.0.0 it can be removed in (x+1).0.0, but if it is deprecated in x.y.z with (y,z) != (0,0) it must not be removed in (x+1).0.0 if (x+1).0.0 is released less than 6 months after x.y.z. This is satisfied by the deprecation rules for comunity.general and community.network. I'm not sure whether official Ansible content satisfies this though - the rules which were once posted somewhere didn't imply this IIRC.)

Reply:
How about the following:
```.rst
* Every ansible minor version series (2.10.x) will keep the major version of a collection constant. So if `community.general` 1.0.0 is included in 2.10.0, 2.10.x will contain the latest `community.general` 1.y.z release that was available when building 2.10.x. If `community.general` 2.0.0 is released, it will **not** be included in ansible 2.10.x, but only in 2.11.0 (if it was published before that was released).
* Therefore, please make sure that the current major release of your collection included in 2.10.0 receives at least bugfixes as long new 2.10.x releases are produced.
* Since new minor releases are included, you can include new features, modules and plugins. You must make sure that you do not break backwards compatibility! (See [semantic versioning](https://semver.org/).) This means in particular:
  * You can fix bugs in patch releases, but not add new features or deprecate things.
  * You can add new features and deprecate things in minor releases, but not remove things or change behavior of existing features.
  * Removals and breaking changes can only be made in major releases.
* Make sure that if a deprecation is added in a collection version that is included in 2.10.x, that the removal itself will only happen in a collection version included in 2.12.0 (or later), but **not** in a collection version included in 2.11.0.
* Content moved from ansible/ansible that was scheduled for removal in 2.11 or later must NOT be removed in the current major release available when ansible 2.10.0 is released. Otherwise it would already be removed in 2.10, unexpectedly for users!
```
(Modulo better wording, that is ;) )

CC @gundalow @abadger @samccann @acozine

Reply:
could we replace `2.11.0` by something more general like 'the next minor ansible following release' ?

Reply:
Done in https://github.com/ansible-collections/overview/pull/94/files